### ISSUE TITLE
Build of docker image fails w/ open file handles, esp. on Windows - [INFO] Caused by: java.io.IOException: request to write '8192' bytes exceeds size in header ...

### ISSUE THREAD

**Issue by scottkurz** [OpenLiberty/boost#131]
As mentioned, this happens especially on Windows.

E.g. you can recreate running the integration test via `mvn install -Pit -Dinvoker.test=*2.0-implicit`

I imagine the fix involves `.dockerignore`...but I haven't looked to understand exactly what we're doing today.

I feel like this might not happen 100% of the time for me but I haven't proven that yet... it might be consistent.

Note the **build.log** in the exc is the **build.log** used to capture the log for the child, IT (maven-invoker) build.

```
[INFO] Caused by: org.apache.maven.plugin.MojoExecutionException: request to write '8192' bytes exceeds size in header of '6793' bytes for entry 'build.log'
[INFO]  at io.openliberty.boost.docker.DockerBuildMojo.execute(DockerBuildMojo.java:106)
[INFO]  at io.openliberty.boost.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:95)
[INFO]  at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
[INFO]  at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
[INFO]  ... 20 more
[INFO] Caused by: java.io.IOException: request to write '8192' bytes exceeds size in header of '6793' bytes for entry 'build.log'
[INFO]  at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.write(TarArchiveOutputStream.java:377)
[INFO]  at java.nio.file.Files.copy(Files.java:2909)
[INFO]  at java.nio.file.Files.copy(Files.java:3069)
[INFO]  at com.spotify.docker.client.CompressedDirectory$Visitor.visitFile(CompressedDirectory.java:303)
[INFO]  at com.spotify.docker.client.CompressedDirectory$Visitor.visitFile(CompressedDirectory.java:253)
[INFO]  at java.nio.file.Files.walkFileTree(Files.java:2670)
[INFO]  at com.spotify.docker.client.CompressedDirectory.create(CompressedDirectory.java:115)
[INFO]  at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1472)
[INFO]  at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1445)
[INFO]  at io.openliberty.boost.docker.DockerBuildMojo.buildDockerImage(DockerBuildMojo.java:154)
[INFO]  at io.openliberty.boost.docker.DockerBuildMojo.execute(DockerBuildMojo.java:104)
[INFO]  ... 23 more
[INFO]  Suppressed: java.io.IOException: This archives contains unclosed entries.
[INFO]          at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.finish(TarArchiveOutputStream.java:220)
[INFO]          at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.close(TarArchiveOutputStream.java:236)
[INFO]          at com.spotify.docker.client.CompressedDirectory.create(CompressedDirectory.java:120)
[INFO]          ... 27 more
```

(I've also seen this when running `boost:docker-build` against a started server, though it's possible that would be hard to completely solve.. not sure.)

**Comment by scottkurz** [comment]
To recreate "against a started server", run ITs, then cd into `boost-maven-plugin/target/it/test-spring-boot-2.0-implicit`, then do a `mvn boost:start` then do a `mvn package`.   The docker build fails for me, on Win 10, with:

```
[ERROR] Failed to execute goal io.openliberty.boost:boost-maven-plugin:0.1.2-SNAPSHOT:docker-build (default) on project test-spring-boot-2.0-implicit: The process cannot access the file because another process has locked a portion of the file -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.openliberty.boost:boost-maven-plugin:0.1.2-SNAPSHOT:docker-build (default) on project test-spring-boot-2.0-implicit: The process cannot access the file because another process has locked a portion of the file
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: The process cannot access the file because another process has locked a portion of the file
        at io.openliberty.boost.docker.DockerBuildMojo.execute(DockerBuildMojo.java:106)
        at io.openliberty.boost.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:95)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 20 more
Caused by: java.io.IOException: The process cannot access the file because another process has locked a portion of the file
        at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
        at sun.nio.ch.FileDispatcherImpl.read(FileDispatcherImpl.java:61)
        at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
        at sun.nio.ch.IOUtil.read(IOUtil.java:197)
        at sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:159)
        at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:65)
        at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:109)
        at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:103)
        at java.io.InputStream.read(InputStream.java:101)
        at java.nio.file.Files.copy(Files.java:2908)
        at java.nio.file.Files.copy(Files.java:3069)
        at com.spotify.docker.client.CompressedDirectory$Visitor.visitFile(CompressedDirectory.java:303)
        at com.spotify.docker.client.CompressedDirectory$Visitor.visitFile(CompressedDirectory.java:253)
        at java.nio.file.Files.walkFileTree(Files.java:2670)
        at com.spotify.docker.client.CompressedDirectory.create(CompressedDirectory.java:115)
        at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1472)
        at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1445)
        at io.openliberty.boost.docker.DockerBuildMojo.buildDockerImage(DockerBuildMojo.java:172)
        at io.openliberty.boost.docker.DockerBuildMojo.execute(DockerBuildMojo.java:104)
        ... 23 more
        Suppressed: java.io.IOException: This archives contains unclosed entries.
                at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.finish(TarArchiveOutputStream.java:220)
                at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.close(TarArchiveOutputStream.java:236)
                at com.spotify.docker.client.CompressedDirectory.create(CompressedDirectory.java:120)
                ... 27 more
```

**Comment by scottkurz** [comment]
Going to reopen until we can test the equivalent with Gradle.   Not going to bother trying until after the current Gradle Docker refactoring is complete.

**Comment by scottkurz** [comment]
The gradle aspect is resolved by #189
